TestStoreRangeMergeNonConsecutive times out after 30sec

The following test appears to have failed:

[#1891](https://circleci.com/gh/cockroachdb/cockroach/1891):

```
I0428 21:34:22.755362     257 multiraft.go:696] New Entry[3]: 6/41 EntryNormal 00000000000000007cd139bff9aef795: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.755546     257 multiraft.go:699] Committed Entry[0]: 6/38 EntryNormal 00000000000000005359ab8439abb31b: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.755729     257 multiraft.go:699] Committed Entry[1]: 6/39 EntryNormal 00000000000000006448d41ce747221f: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.756024     257 multiraft.go:699] Committed Entry[2]: 6/40 EntryNormal 000000000000000017856db6a1a1e735: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.756186     257 multiraft.go:699] Committed Entry[3]: 6/41 EntryNormal 00000000000000007cd139bff9aef795: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
panic: test timed out after 30s

goroutine 3084 [running]:
testing.func·008()
	/usr/src/go/src/testing/testing.go:681 +0x12f
created by time.goFunc
	/usr/src/go/src/time/sleep.go:129 +0x4b

goroutine 1 [chan receive]:
testing.RunTests(0x1075bc8, 0x15112a0, 0xa1, 0xa1, 0xc207ffb901)
	/usr/src/go/src/testing/testing.go:556 +0xad6
--
	/go/src/github.com/cockroachdb/cockroach/storage/range.go:854 +0x212
github.com/cockroachdb/cockroach/util.func·005()
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:73 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:74 +0xe3
FAIL	github.com/cockroachdb/cockroach/storage	30.019s
=== RUN TestBatchBasics
I0428 21:35:01.959947     352 rocksdb.go:88] opening in-memory rocksdb instance
I0428 21:35:01.960823     352 rocksdb.go:112] closing rocksdb instance at ""
--- PASS: TestBatchBasics (0.00s)
=== RUN TestBatchGet
I0428 21:35:01.961571     352 rocksdb.go:88] opening in-memory rocksdb instance
I0428 21:35:01.961818     352 rocksdb.go:112] closing rocksdb instance at ""
--- PASS: TestBatchGet (0.00s)
=== RUN TestBatchMerge
I0428 21:35:01.962618     352 rocksdb.go:88] opening in-memory rocksdb instance
I0428 21:34:22.755362     257 multiraft.go:696] New Entry[3]: 6/41 EntryNormal 00000000000000007cd139bff9aef795: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.755546     257 multiraft.go:699] Committed Entry[0]: 6/38 EntryNormal 00000000000000005359ab8439abb31b: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.755729     257 multiraft.go:699] Committed Entry[1]: 6/39 EntryNormal 00000000000000006448d41ce747221f: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.756024     257 multiraft.go:699] Committed Entry[2]: 6/40 EntryNormal 000000000000000017856db6a1a1e735: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0428 21:34:22.756186     257 multiraft.go:699] Committed Entry[3]: 6/41 EntryNormal 00000000000000007cd139bff9aef795: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
panic: test timed out after 30s

goroutine 3084 [running]:
testing.func·008()
	/usr/src/go/src/testing/testing.go:681 +0x12f
created by time.goFunc
	/usr/src/go/src/time/sleep.go:129 +0x4b

goroutine 1 [chan receive]:
testing.RunTests(0x1075bc8, 0x15112a0, 0xa1, 0xa1, 0xc207ffb901)
	/usr/src/go/src/testing/testing.go:556 +0xad6
--
	/go/src/github.com/cockroachdb/cockroach/storage/range.go:854 +0x212
github.com/cockroachdb/cockroach/util.func·005()
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:73 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:74 +0xe3
FAIL	github.com/cockroachdb/cockroach/storage	30.019s
=== RUN TestBatchBasics
I0428 21:35:01.959947     352 rocksdb.go:88] opening in-memory rocksdb instance
I0428 21:35:01.960823     352 rocksdb.go:112] closing rocksdb instance at ""
--- PASS: TestBatchBasics (0.00s)
=== RUN TestBatchGet
I0428 21:35:01.961571     352 rocksdb.go:88] opening in-memory rocksdb instance
I0428 21:35:01.961818     352 rocksdb.go:112] closing rocksdb instance at ""
--- PASS: TestBatchGet (0.00s)
=== RUN TestBatchMerge
I0428 21:35:01.962618     352 rocksdb.go:88] opening in-memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_merge_test.go
+++ b/storage/client_merge_test.go
@@ -232,6 +232,12 @@ func TestStoreRangeMergeNonConsecutive(t *testing.T) {
 	// This is sufficient for now to generate the "ranges not collocated" error; if this changes
 	// in the future we could make this test more realistic by using a multiTestContext
 	// and ChangeReplicas to arrange two ranges onto different stores/nodes.
+	//
+	// Wait for the leader lease to ensure things are quiescent before removing the range.
+	// See #702 and TestStoreExecuteCmdOutOfRange.
+	// TODO(bdarnell): refactor this test to rebalance the range onto a separate node
+	// when this is supported.
+	rangeB.WaitForLeaderLease(t)
 	if err := store.RemoveRange(rangeB); err != nil {
 		t.Fatal(err)
 	}

